lib/submit-analytics.js

```diff
@@ -33,50 +117,158 @@ function sha256(data) {
+                'Anonymized Machine Identifier': identifier,
+                'Anonymized Application ID': sha256(__dirname),
+                'Binding': environment,
+                'Version': context.version,
+                'Language': 'javascript',
+                'OS Type': os.platform(),
+                'OS Version': os.release(),
+                'Node.js versions': process.versions
+            }
+        }
+    };
 
-            machineId()
-                .catch() // Ignore errors
-                .then((identifier) => {
-                    if (!identifier) {
-                        identifier = sha256('unknown');
-                    }
-
-                    const payload = {
-                        'event': eventName,
-                        'properties': {
-                            'token': 'aab85907a13e1ff44a95be539d9942a9',
-                            'distinct_id': identifier,
-                            'Anonymized Machine Identifier': identifier,
-                            'Anonymized Application ID': sha256(__dirname),
-                            'Binding': context || 'node.js',
-                            'Version': require('../package.json').version,
-                            'Language': 'javascript',
-                            'OS Type': os.platform(),
-                            'OS Version': os.release(),
-                            'Node.js versions': process.versions
-                        }
-                    };
-
-                    request(`https://api.mixpanel.com/track/?data=${Buffer.from(JSON.stringify(payload), 'utf8').toString('base64')}&ip=1`,
-                    () => { /* Analytics failed. Do nothing. */ });
-                });
-        }, 100);
+    return payloads;
+}
+
+/**
+ * Generate and submit analytics data to servers.
+ * @param  {[WebhookProdDetails|WebhookStageDetails|MixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Promise} Promise that will resolve if analytics are successfully submitted
+ */
+async function submitAnalytics(requestFactory, context, eventName) {
+    // delay execution by 100ms
+    // the caller mat set REALM_DISABLE_ANALYTICS in this time
+    // to disable analytics
+    await new Promise(resolve => setTimeout(resolve, 100));
+    if (isAnalyticsDisabled()) {
+        return;
     }
 
-    if (require.main === module) {
-        module.exports('Install');
+    const payload = await fetchPlatformData(context, eventName)
+    const versionTags = payload['mixPanel']['properties']['Version'].split('.');
+
+    if (versionTags[0] < 10) {
+        // on v6.x, we submit both MixPanel and S3 analytics
+        // MixPanel analytics make no distinction between production and staging modes
+        await dispatchAnalytics(new MixpanelDetails(), payload);
```

Comment:
What happens when this request eventually fail (when the MixPanel subscription ends)? Perhaps a `Promise.all([ ... ])` or a `try {} finally {}` block to allow the S3 to dispatch even if the MixPanel request fail?